tests/unit_tests/cuda/mask_store_cuda.cpp

```diff
@@ -0,0 +1,111 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <climits>
+#include <iostream>
+#include <vecmem/memory/cuda/managed_memory_resource.hpp>
+
+#include "mask_store_cuda_kernel.hpp"
+
+TEST(mask_store_cuda, mask_store) {
+
+    // memory resource
+    vecmem::cuda::managed_memory_resource mng_mr;
+
+    // Types must be sorted according to their id (here: masks/mask_identifier)
+    mask_store<vecmem::vector, rectangle, trapezoid, ring, cylinder, single,
+               annulus>
+        store(mng_mr);
+
+    ASSERT_TRUE(store.template empty<e_annulus2>());
+    ASSERT_TRUE(store.template empty<e_cylinder3>());
+    ASSERT_TRUE(store.template empty<e_rectangle2>());
+    ASSERT_TRUE(store.template empty<e_ring2>());
+    ASSERT_TRUE(store.template empty<e_single3>());
+    ASSERT_TRUE(store.template empty<e_trapezoid2>());
+
+    store.template add_mask<e_rectangle2>(1.0, 2.0);
+    store.template add_mask<e_trapezoid2>(0.5, 1.5, 4.0);
+    store.template add_mask<e_ring2>(1.0, 10.0);
+    store.template add_mask<e_cylinder3>(1., 0.5, 2.0);
+    // store.group<e_single3>().push_back(single{3.0,6.0});
+    store.template add_mask<e_annulus2>(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0);
+
+    ASSERT_FALSE(store.template empty<e_annulus2>());
+    ASSERT_FALSE(store.template empty<e_cylinder3>());
+    ASSERT_FALSE(store.template empty<e_rectangle2>());
+    ASSERT_FALSE(store.template empty<e_ring2>());
+    // ASSERT_FALSE(store.template empty<e_single3>());
+    ASSERT_FALSE(store.template empty<e_trapezoid2>());
+
+    /** Generate random points for test **/
+    vecmem::vector<point2> input_point2(n_points, &mng_mr);
+    vecmem::vector<point3> input_point3(n_points, &mng_mr);
+
+    for (int i = 0; i < n_points; i++) {
+        point2 rand_point2 = {rand() % 100 / 10., rand() % 100 / 10.};
```

Comment:
Is this ```std::rand``` from ```<cstdlib>```?

Reply:
I don't think I included it explicitly, but it should be from <cstdlib> which might be included somewhere-else in one of dependenies

Reply:
I think its good practice to include things directly and name them explicitly

Reply:
done